doc/ondemand.md

```diff
@@ -421,30 +453,46 @@ mutable reference to it.
 
 ### Iteration Safety
 
-  - If the value fails to be parsed as one type, you can try to parse it as something else until you
-    succeed.
-  - Safety: If the value succeeds in being parsed or converted to a type, you cannot try again. (It
-    sets `json` to null, so you will segfault.) This prevents double iteration of an array (which
-    will cause inconsistent iterator state) or double-unescaping a string (which could cause memory
-    overruns if done too much).
+The On Demand API is powerful. To compensate, we add some safeguards to ensure that it can be used without fear
+in production systems:
+
+  - If the value fails to be parsed as one type, the program can try to parse it as something else until the program succeeds. Thus 
+    the programmer can engineer fall back routines.
+  - If the value succeeds in being parsed or converted to a type, the program cannot try again. An attempt to parse the same node twice will
+    cause the program to abort. We put this safety measure in the API to prevent double iteration of an array which
+    would cause inconsistent iterator state or double-unescaping a string which may cause memory
+    overruns if done.
   - Guaranteed Iteration: If you discard a value without using it--perhaps you just wanted to know
-    if it was null but didn't care what the actual value was--it will iterate. The destructor will
-    take care of this.
+    if it was `nullptr` but did not care what the actual value was--it will iterate. The destructor automates
+    the iteration.
+
+### Applicability and Limitations of the On Demand Approach
+
+We expect that the On Demand approach has many of the performance benefits of the schema-based approach, while providing a flexibility that is similar to that of the DOM-based approach. However, there are some limitations.
+
+Pros of the On Demand approach:
+* Faster than DOM in some cases. Reduced memory usage.
+* Straightforward, programmer-friendly interface (arrays and objects).
 
-### Raw Key Lookup
+Cons of the On Demand approach:
+* Because it operates in streaming mode, you only have access to the current element in the JSON document. Furthermore, the document is traversed in order so the code is sensitive to the order of the JSON nodes in the same manner as an event-based approach (e.g., SAX). It is possible for the programmer to handle out-of-order keys, but it requires additional care. Yiou should be mindful that the though your software might write the keys in a consistent manner, the JSON specification does not prescribe that the order be significant and thus, a JSON producer could change the order of the keys within an object. The On Demand API will still help the programmer by throwing an exception when the unexpected occurs, but the programmer is responsible for handling such cases (e.g., by rejecting the JSON input).
```

Comment:
Yiou -> You

Reply:
Fixed that an a poorly worded new sentence elsewhere.